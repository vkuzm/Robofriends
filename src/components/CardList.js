import React from 'react';
import Card from '../components/Card';

const CardList = ({ robots }) => {
	//if (true) {
	//	throw new Error('Nooooo');
	//}

	const cardList = robots.map(robot => {
	  return (
	    <Card 
		    key={robot.id} 
		    id={robot.id} 
		    name={robot.name} 
		    email={robot.email}
		  />
	  );
	})

	return <div>{cardList}</div>;
}

export default CardList;