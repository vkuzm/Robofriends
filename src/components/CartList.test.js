import React from 'react';
import CardList from './CardList';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

it('expect to render CardList shallowComponent', () => {
  const mockRobots = [
  	{
  		id: 1,
  		name: 'R2D2',
  		email: 'r2d2@mail.com'
  	},
  	{
  		id: 2,
  		name: '3P20',
  		email: '3p20@mail.com'
  	},
  	{
  		id: 3,
  		name: 'R8S2',
  		email: 'r8s2@mail.com'
  	}
  ]

	const shallowComponent = toJson(shallow(<CardList robots = {mockRobots} />));
  expect(shallowComponent).toMatchSnapshot();
});