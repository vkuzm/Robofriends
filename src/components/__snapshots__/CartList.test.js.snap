// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList shallowComponent 1`] = `
<div>
  <Card
    email="r2d2@mail.com"
    id={1}
    key="1"
    name="R2D2"
  />
  <Card
    email="3p20@mail.com"
    id={2}
    key="2"
    name="3P20"
  />
  <Card
    email="r8s2@mail.com"
    id={3}
    key="3"
    name="R8S2"
  />
</div>
`;
